test/jdk/java/awt/FullScreen/NoResizeEventOnDMChangeTest/NoResizeEventOnDMChangeTest.java

```diff
@@ -178,6 +189,10 @@ public void run() {
             });
         } catch (Exception ex) {}
 
+        if (skipTest.get()) {
+            // Skipping test because graphics driver switched window to another screen
+            return;
```

Comment:
This test isn't about multi-mon .. it is about full screen.
If the window doesn't go full screen on the display we expect that's odd.
Are you saying it went FS on one display and then jumped ?
Can't we make the test more robust about it going on the display we expect OR pivot the test to verifying on the display that went FS ?

Reply:
> Are you saying it went FS on one display and then jumped ?

That is exactly what happened and only in this case the test were failed - obviously, because the window was on a secondary display and resize was happening on the primary so window has not got any events about resize. I added a lot of debug information to the native code but after that i ran test for hours non stop and it does not reproduce - i guess it is all due to the timings. You see, when we changing resolution the physical display goes offline (at least on my system with NVidia card) meaning status LED goes from green to amber, then it goes green again with the new resolution set. If request to get full screen windows from GD comes at that exact moment the window might be placed on a secondary display which does not change. I do not know how to force OS not to reassign window to another display in this case.

Reply:
In the scenario above what will happen if the window is jumped from one screen to another? Will it be full screen or it will be in the normal mode? Which (old or new) graphics device will report that the window is full screen on it? I guess we should move the window back to the old device in full-screen mode, or we should move the window to the new device but in the normal mode, in both cases, the resize event should be generated, isn't it?

Reply:
> In the scenario above what will happen if the window is jumped from one screen to another? Will it be full screen or it will be in the normal mode? Which (old or new) graphics device will report that the window is full screen on it? I guess we should move the window back to the old device in full-screen mode, or we should move the window to the new device but in the normal mode, in both cases, the resize event should be generated, isn't it?

It opens up as a full screen window on the secondary display, i can't check which device reports it because when i add debug output i can not reproduce the failure. If two displays have different resolution then moving window from one GD to another it should generate the resize event on both window and the component inside it.

Reply:
Then you can check that the window jumped to a different screen and the size counter should not be increased if the requested mode and the size of the new screen are the same. You do not need to skip the whole test in this case. But it also looks like a bug that the window is jumped to the other screen and is not returned back.

Reply:
> Then you can check that the window jumped to a different screen and the size counter should not be increased if the requested mode and the size of the new screen are the same.

Ok, done exactly that.

> But it also looks like a bug that the window is jumped to the other screen and is not returned back.

Yes but that might be the issue with the Windows itself and i do not think there is anything that can be done on this test level.